src/formatDate.js

```diff
@@ -9,6 +9,62 @@
  */
 function formatDate(date, fromFormat, toFormat) {
   // write code here
+  const outputNeeded = {
+    day: '',
+    month: '',
+    year: '',
+    separator: '',
+  };
+  const separatedDate = date.split(fromFormat[fromFormat.length - 1]);
+  const result = [];
+
+  // check inData and start to build object for the result creating
+  for (let i = 0; i < fromFormat.length; i++) {
+    const item = fromFormat[i];
+
+    switch (item[0].toLowerCase()) {
+      case 'd':
+        outputNeeded.day = `${separatedDate[i]}`;
+        break;
+      case 'm':
+        outputNeeded.month = `${separatedDate[i]}`;
+        break;
+      case 'y':
+        const yearIn =
+          item.length === 2 && Number(separatedDate[i]) < 30
+            ? `20${separatedDate[i].slice(-2)}`
+            : `19${separatedDate[i].slice(-2)}`;
+
+        outputNeeded.year = item.length === 2 ? yearIn : `${separatedDate[i]}`;
+        break;
+      default:
+        outputNeeded.separator = `${toFormat[i]}`;
+    }
+  }
+
+  // check output conditions and build our result
+  for (let i = 0; i < toFormat.length - 1; i++) {
+    const item = toFormat[i];
+
+    switch (item[0].toLowerCase()) {
+      case 'd':
+        result.push(outputNeeded.day);
+        break;
+      case 'm':
+        result.push(outputNeeded.month);
+        break;
+      case 'y':
+        const year =
+          item.length < 4 ? outputNeeded.year.slice(-2) : outputNeeded.year;
```

Comment:
The logic to convert the year from 'YYYY' to 'YY' is incorrect. You should check the length of the `item` before slicing the year. If `item.length === 4`, then you should push the full year; otherwise, if `item.length === 2`, push only the last two digits.